HeterogeneousCore/CUDAUtilities/test/copyAsync_t.cpp

```diff
@@ -21,8 +21,9 @@ TEST_CASE("copyAsync", "[cudaMemTools]") {
       auto host = cudautils::make_host_unique<int>(stream);
 
       cudautils::copyAsync(device, host_orig, stream);
-      cuda::memory::async::copy(host.get(), device.get(), sizeof(int), stream);
       cudaCheck(cudaStreamSynchronize(stream));
+      cudaMemcpyAsync(host.get(), device.get(), sizeof(int), cudaMemcpyDeviceToHost, stream);
+      stream.synchronize();
```

Comment:
These two lines seem incorrect:
  - `cudaMemcpyAsync` should come before `cudaStreamSynchronize`
  - `stream.synchronize();` seems spurious, possibly due to incorrect conflict resolution